zws40/Intro_to_CodeChat.c

```diff
@@ -0,0 +1,140 @@
+// ====================
+
+// This document will walk you through solving the Mars Lander puzzle on `codingame.com <https://www.codingame.com/ide/puzzle/mars-lander-episode-1>`_
+
+// .. image:: mars_lander_ss_png.png
+
+// The puzzle focuses on the use of conditions in the C language
+
+
+// Press `here <https://tutorialtpoint.in/c/conditional-statements-in-c/>`_ for a review of conditional statements in C
+
+// =========================
+// Thoughts Before Coding
+// =========================
+
+// This game has you controlling a lander to safely land it on mars. Using the lander's speed, thrust, and angle, the user must code an algorithm to land on Mars without exploding
+
+// For the puzzle we are given seven integer type variables
+
+// 1. X coordinate of the rocket
+// 2. Y coordinate of the rocket
+// 3. Horizontal Speed
+// 4. Vertical Speed
+// 5. Fuel Level
+// 6. Angle of Rotation
+// 7. Thrust Power
+
+// **Things to consider**
+
+// * How quickly will the fuel run out?
+// * At what elevation does speed need to be sharply reduced?
+// * What is the maximum speed the lander can have and still make a safe landing (Safe meaning not blowing up on impact)
+//
+
+// #. Fuel will constantly be used as you thrust upwards. If you do not get landed fast enough, you will run out of fuel and crash into the ground
+// #. The elevation and speed are tightly linked. You MUST make sure the lander is not traveling to quickly when low in elevation, but also must have enough speed to reach the ground before running out of fuel
+
+// * The fuel is very limited, limited enough that if you spend to much time not on the ground, you will lose all thrust and slam into the ground
+// * The elevation is key for speed control. Once at a certain speed at a certain elevation, the full force thrust of the lander simply will not slow you down enough to stop from crashing into the ground 
+
+// *For this puzzle, all we need to worry about is the Y coordinate of the rocket, the vertical speed, the fuel level, and the thruster power.*
+//
+
+// Only paying attention to your Y coordinate, the vertical speed, the fuel level, and the thruster power is because of the level of Mars Lander. Later in the series with part two you have more variables due to the harsher landing conditions. In this level it is a completely flat landing area so not as advanced.
+
+// =========================
+// Implementing Code
+// =========================
+
+// **All code below down to the next bold message is pre generated code for the game**
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <stdbool.h>
+
+
+// Auto-generated code below aims at helping you parse
+// the standard input according to the problem statement.
+
+
+int main()
+{    
+    // the number of points used to draw the surface of Mars.
+    int surface_n;
+    scanf("%d", &surface_n);
+    for (int i = 0; i < surface_n; i++) {
+    // X coordinate of a surface point. (0 to 6999)
```

Comment:
Indent this comment, since it belongs to the code below it. Also remove the blank line so comment and code are clearly connected.